packages/fetch/src/cache.ts

```diff
@@ -89,6 +91,37 @@ export const withCache: RequestModifier =
     });
   };
 
+export const withRefetchOnExpiry: RequestModifier = 
+  <Result extends unknown, FetcherArgs extends any[]>(pollDelayMs = 100) =>
+  (requestContext: RequestContext<Result, FetcherArgs>) => {
+    wrapFetcher<Result, FetcherArgs>(
+      requestContext,
+      <T>(originalFetcher: any) =>
+        (requestData, info) => {
+          const serializedRequest = serializeRequest(requestData);
+          const cached: CacheEntry | undefined = requestContext.cache[serializedRequest];
+          if (!cached) {
+            if (typeof requestContext.expires === 'number') {
+              setTimeout(() => requestContext.resource?.[1].refetch(), requestContext.expires + 10);
+            } else if (typeof requestContext.expires === 'function') {
+              const poll = () => {
+                if (requestContext.expires(cached)) {
+                  requestContext.resource?.[1].refetch();
+                } else if (pollDelayMs === 0) {
+                  requestAnimationFrame(poll);
```

Comment:
Why raf and not setTimeout? Is it because it'll happen sooner?

Reply:
The minimal resolution is ~16ms anyways, but raf will have slightly better performance, because it will always trigger between the client render cycles.